modules/products/products.service.js

```diff
@@ -0,0 +1,77 @@
+const Products = require('./products.model');
+const Size = require('../../models/Size');
+
+class ProductsService {
+  getAllProducts() {
+    return Products.find();
+  }
+
+  getProductsById(id) {
+    return Products.findById(id);
+  }
+
+  getSizeById(id) {
+    return Size.findById(id);
+  }
+
+  filterItems(args = {}){
+    const filter = {};
+    const {
+      pattern, colors, price
+    } = args;
+  
+    if (colors.length) {
+      filter.colors = {
+        $elemMatch: {
+          simpleName: { $in: colors },
+        },
+      };
+    }
+    if (pattern.length) {
+      filter.pattern = {
+        $elemMatch: {
+          value: { $in: pattern },
+        },
+      };
+    }
+    if(price){
+      filter.basePrice = {
+          $gte: price[0], 
+          $lte: price[1]
+      };
+    }
+    return filter;
+  };
+
+  async getProductsByOptions({filter, skip, limit, sort, search}) {
+    
+    const isNotBlank = str => !(!str || str.trim().length === 0);
```

Comment:
Do you need a separate function if it is used only once?

Reply:
it is easier to read the code this way